import java.io.IOException;

public class Ex03_Exception_finally {

	static void startInstall() {
		System.out.println("INSTALL");
	}
	
	static void copyFiles() {
		System.out.println("COPY FILES");
	}
	
	static void fileDelete() {
		System.out.println("DELETE FILES");
	}
	
	public static void main(String[] args) {

		try {
			copyFiles();
			startInstall();
			
			//개발자가 임의로 예외를 만들 수 있다.
			//A, B, C 정상입력,, D라고 입력하면 비정상으로 하고 싶다..
			// => 특수한 경우를 예외로 만들 수 있다.
			// => ==사용자 예외 던지기 (예외객체 생성 > 예외객체를 개발자가 직접 생성(new))
			
			//throw new IOException("Install 처리 중 문제 발생"); //필요한 객체 만들어서 던지기 
			//*throw new IOException()
			
		} catch(Exception e) {
			System.out.println("예외 메세지 출력: "+e.getMessage());
		} finally {
			//예외가 발생하던, 발생하지 않던 강제적 실행 블록
			fileDelete();
		}
		
		System.out.println("MAin End");
		
	}

}
